kuma/landing/views.py

```diff
@@ -49,6 +52,41 @@ def promote_buttons(request):
     return render(request, 'landing/promote_buttons.html')
 
 
+@never_cache
+def contribute(request):
+    initial_data = {}
+    if request.user and request.user.is_authenticated() and request.user.email:
+        initial_data = {'email': request.user.email}
+
+    if request.POST:
+        form = ContributionForm(request.POST)
+        if form.is_valid():
+            charge = form.make_charge()
+            if charge and charge.id and charge.status == 'succeeded':
+                if settings.MDN_CONTRIBUTION_CONFIRMATION_EMAIL:
+                    contribute_thank_you_email.delay(
+                        form.cleaned_data['name'],
+                        form.cleaned_data['email']
+                    )
+                return redirect('contribute_confirmation_succeeded')
+            return redirect('contribute_confirmation_error')
+
+        form = ContributionForm(request.POST)
```

Comment:
Why is the form reinitialized here? I think this throws away any form validation errors.

Reply:
@michal-macioszczyk maybe you can look into this?